### ISSUE TITLE
Create basic architecture of the lib

### ISSUE THREAD

**Issue by mishani0x0ef** [mishani0x0ef/not-react#1]
Before any work has started I must define basic concepts of how future lib should work and what to expect from it.

It's a kind of exploratory task that should give me a vision of future work.

- [x] Setup wiki spaces
- [ ] Create architecture and design section
- [ ] Describe high-level flow of lib
- [ ] Define base building blocks of the architecture
- [ ] Define responsibilities
- [ ] Provide scheme of interaction between components
- [ ] Define next steps
